### PR TITLE
Added SessionFactoryCreator and related helpers

### PR DESCRIPTION
**Pull request by devinrsmith**
This is the java client related changes to support #5374. Note, that we may need to prioritize https://github.com/deephaven/web-client-ui/issues/1947 as follow-up.

### CODE REVIEW

**Review by nbauernfeind — APPROVED** [review]

File: java-client/session/src/main/java/io/deephaven/client/impl/ClientConfig.java [hunk 1]
```diff
@@ -58,6 +58,27 @@ public int maxInboundMessageSize() {
...
+     */
+    public ClientConfig withSsl(SSLConfig sslConfig) {
+        if (sslConfig.equals(ssl().orElse(null))) {
+            return this;
+        }
+        final Builder builder = builder()
+                .target(target())
+                .ssl(sslConfig)
+                .putAllExtraHeaders(extraHeaders())
+                .maxInboundMessageSize(maxInboundMessageSize());
```
Comment (rcaudy) [inline java-client/session/src/main/java/io/deephaven/client/impl/ClientConfig.java:76]: With copy?

File: server/src/main/java/io/deephaven/server/barrage/BarrageSessionFactoryClient.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                .allocator(allocator)
+                .authenticationTypeAndValue(authenticationTypeAndValue)
+                .build();
+    }
+
+    /**
+     * Provides an application id as {@link BarrageSessionFactoryClient} class name. A
+     * {@link BarrageSessionFactoryClient} is set as the field name {@value INSTANCE}.
+     */
+    public static final class Application implements Factory {
```
[the remark this answers is not in the document]
Comment (devinrsmith) [reply server/src/main/java/io/deephaven/server/barrage/BarrageSessionFactoryClient.java]: I've removed it as an application.

**Comment by devinrsmith** [comment]
I've labelled as breaking since BarrageSessionFactoryBuilder was removed as extraneous fluff; all callers should be able to use `DaggerDeephavenBarrageRoot.create().factoryBuilder()`. I've verified w/ Andy that DHE does _not_ extend the interface.

**Review by nbauernfeind — APPROVED** [review]

**Comment by devinrsmith** [comment]
All nightlies pass

**Pull request merged**